lib/api/controllers/documentController.js

```diff
@@ -204,6 +275,69 @@ function DocumentController (kuzzle) {
   };
 
   /**
+   * Delete specific documents according to given ids
+   * Delegates the notification to delete action
+   *
+   * @param {Request} request
+   * @returns {Promise<Object>}
+   */
+  this.mDelete = function documentMDelete (request) {
+    let promises = [];
+
+    assertHasBody(request, 'document:mDelete');
+    assertBodyHasAttribute(request, 'ids', 'document:mDelete');
+    assertHasIndexAndCollection(request, 'document:mDelete');
+
+    if (!Array.isArray(request.input.body.ids)) {
+      throw new BadRequestError('document:mDelete must specify an array of "ids".');
+    }
+
+    for (let i = 0; i < request.input.body.ids.length; i++) {
+      let deleteRequest = new Request({
+        index: request.input.resource.index,
+        collection: request.input.resource.collection,
+        controller: 'document',
+        action: 'delete',
+        _id: request.input.body.ids[i]
+      }, request.context);
+
+      promises[i] = new Promise(resolve => {
+        kuzzle.funnel.processRequest(deleteRequest)
+          .then(() => {
+            return resolve(deleteRequest);
+          })
+          .catch(error => {
+            deleteRequest.setError(error);
+
+            return resolve(deleteRequest);
+          });
+      });
+    }
+
+    return Promise.all(promises)
+      .then(documentsRequests => {
+        let
+          deletedIds = [],
+          errorMessages = [];
+
+        for (let i = 0; i < documentsRequests.length; i++) {
+          if (documentsRequests[i].error || documentsRequests[i].result.result !== 'deleted') {
+            errorMessages.push('document:mDelete was not able to delete the document ' + documentsRequests[i].input.resource._id);
+          }
+          else {
+            deletedIds.push(documentsRequests[i].input.resource._id);
+          }
+        }
+
+        if (errorMessages.length > 0) {
+          request.setError(new PartialError('document:mDelete was not able to remove all documents', errorMessages.join('; ')));
```

Comment:
second argument of the constructor of [PartialError](https://github.com/kuzzleio/kuzzle-common-objects/blob/master/lib/errors/partialError.js) is an array of errors